### PR TITLE
Distributed training

### PR DESCRIPTION
**Pull request by chester-leung**

### CODE REVIEW

**Comment by trivialfis** [comment]
?

**Comment by chester-leung** [comment]
Apologies. Did not mean to PR to this repository.

**Comment by hcho3** [comment]
@trivialfis https://github.com/mc2-project/mc2/tree/master/federated-xgboost This project seems quite interesting.

**Comment by trivialfis** [comment]
Oh yeah we looked into it.  If I'm not mistaken it adds an additional layer on top of Unix socket.  In the context of XGBoost, it can prevent Eve from spying on the transferred model, and CPU gradients.
